authors/apps/articles/views.py

```diff
@@ -57,3 +61,72 @@ def destroy(self, request, pk, **kwargs):
                 return Response("You have succesfully deleted the article")
             return Response("You do not have permision to delete the article")
         return Response(self.err_message, status=status.HTTP_404_NOT_FOUND)
+
+
+class Like(APIView):
+    permission_classes = (IsAuthenticated, )
+
+    @classmethod
+    def post(self, request, **kwargs):
+        current_user = request.user.username
+        article = Article.objects.filter(id=kwargs['article_id'])
+        if article:
+            my_like_already = Likes.objects.filter(
+                article_id=kwargs['article_id'], reviewer_id=current_user, like_or_dislike=1)
+            if not my_like_already:
+                data = {}
+                data['reviewer'] = current_user
+                data['like_or_dislike'] = 1
+                data['article'] = article.first().id
+                serializer = LikeSerializer(data=data)
```

Comment:
@KabohaJeanMark Why the need to create a dictionary. Then serialise, validate it and  save the data into the database. Yet the data you are saving to the database is created within the system itself and not being passed to it from the outside. Don't you think all this is all unnecessary.

Reply:
Yes @archbishop. You were right. A simple object creation has sufficed